Testscripts/Windows/VERIFY-XDP-LOAD-UNLOAD.ps1

```diff
@@ -0,0 +1,183 @@
+    }
+    Write-LogInfo "XDPDump application ran successfully for count: $LogSuffix"
+    return "PASS"
+}
+
+function Main {
+    try {
+        $noClient = $true
+        $noServer = $true
+        foreach ($vmData in $allVMData) {
+            if ($vmData.RoleName -imatch "receiver") {
+                $clientVMData = $vmData
+                $noClient = $false
+            }
+            elseif ($vmData.RoleName -imatch "sender") {
+                $noServer = $false
+                $serverVMData = $vmData
+            }
+        }
+        if ($noClient) {
+            Throw "No any receiver VM defined. Aborting Test."
+        }
+        if ($noServer) {
+            Throw "No any sender VM defined. Aborting Test."
+        }
+
+        #CONFIGURE VM Details
+        Write-LogInfo "CLIENT VM details :"
+        Write-LogInfo "  RoleName : $($clientVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($clientVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($clientVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($clientVMData.InternalIP)"
+        Write-LogInfo "SERVER VM details :"
+        Write-LogInfo "  RoleName : $($serverVMData.RoleName)"
+        Write-LogInfo "  Public IP : $($serverVMData.PublicIP)"
+        Write-LogInfo "  SSH Port : $($serverVMData.SSHPort)"
+        Write-LogInfo "  Internal IP : $($serverVMData.InternalIP)"
+
+        # Check for compatible kernel
+        $currentKernelVersion = Run-LinuxCmd -ip $clientVMData.PublicIP -port $clientVMData.SSHPort `
+            -username $user -password $password -command "uname -r"
+        # ToDo: Update Minimum kernel version check once patches are in downstream distro.
+        if ((Compare-KernelVersion $currentKernelVersion $MIN_KERNEL_VERSION) -lt 0 -or $global:DetectedDistro -ne "UBUNTU"){
+            Write-LogInfo "Minimum kernel version required for XDP: $MIN_KERNEL_VERSION."`
+                "Unsupported kernel version: $currentKernelVersion or Unsupported distro: $($global:DetectedDistro)."
+            return $global:ResultSkipped
+        }
+
+        # PROVISION VMS FOR LISA WILL ENABLE ROOT USER AND WILL MAKE ENABLE PASSWORDLESS AUTHENTICATION ACROSS ALL VMS.
+        Provision-VMsForLisa -allVMData $allVMData -installPackagesOnRoleNames "none"
+
+        # Generate constants.sh and write all VM info into it
+        Write-LogInfo "Generating constants.sh ..."
+        $constantsFile = "$LogDir\constants.sh"
+        Set-Content -Value "# Generated by Azure Automation." -Path $constantsFile
+        Add-Content -Value "ip=$($clientVMData.InternalIP)" -Path $constantsFile
+        Add-Content -Value "nicName=eth1" -Path $constantsFile
+        foreach ($param in $currentTestData.TestParameters.param) {
+            Add-Content -Value "$param" -Path $constantsFile
+        }
+        Write-LogInfo "constants.sh created successfully..."
+        Write-LogInfo (Get-Content -Path $constantsFile)

```

Comment:
why do we need it?

Reply:
To cross check content of constants.sh as it is auto-generated.